File: C++/Hard/WrongAnswer/naming-a-company_8.cpp
class Solution {
public:
    long long distinctNames(vector<string>& ideas) {
        
        int n = ideas.size();
        unordered_map<string, int> m;
        for (auto& x : ideas) m[x] = 1;
        long long ans = 0;
        for (int i = 0; i < n; ++i) {
            for (int j = i + 1; j < n; ++j) {
                string a = ideas[i], b = ideas[j];
                swap(a[0], b[0]);
                if (m[a] == 0 && m[b] == 0) {
                    ans++;
                }
            }
        }
        return ans;
    }
};

File: Java/Hard/WrongAnswer/naming-a-company_8.java
class Solution {
    public long distinctNames(String[] ideas) {
        
        Set<String> set = new HashSet<>();
        for (String idea : ideas) {
            for (String other : ideas) {
                if (idea.equals(other)) {
                    continue;
                }
                String newName = other.substring(0, 1) + idea.substring(1) + " " + idea.substring(0, 1) + other.substring(1);
                set.add(newName);
            }
        }
        return set.size();
    }
}